web/components/templates/hql/QueryResult.tsx

```diff
@@ -53,14 +55,6 @@ function QueryResult({
     );
   }
 
-  if (loading) {
-    return (
-      <div className="p-4 text-center text-muted-foreground">
-        <LoadingAnimation />
-      </div>
-    );
-  }
-
   const columns = Object.keys(result[0]);
```

Comment:
can we wrap this in a useMemo

```suggestion
  const columns = useMemo(() => Object.keys(result[0]), result);
```